linode/manager
# refactor: [M3-7397] - Clean up `App.tsx`

## Description 📝

`App.tsx` has gotten very messy and generally  hard to consume. This PR cleans up `App.tsx` and related pieces. Most changes are just moving code. The largest refactoring happens in `useToastNotifications`.

## Changes  🔄
- Cleans up and improves `ToastNotifications` and makes it `useToastNotifications` (inspired by https://github.com/linode/manager/pull/9416)
- Created `useAdobeAnalytics`, `useNewRelic`, `useEventHandlers`, and `useGlobalKeyboardListener` to separate out logic

## How to test 🧪

### Verification steps 
- Perform a general check of Cloud Manager
- Verify global toast notifications still work (creating or deleting a volume for example)
- Verify that both Adobe Analytics and New Relic initialize
  - This can be verified by looking at the preview link and inspecting the network requests. Verify you see requests to both adobe and New Relic
- Verify keyboard shortcuts like `Control` + `Shift` + `D` still work 
- Verify e2es pass

## As an Author I have considered 🤔
